crates/e2e/tests/e2e/refunder.rs

```diff
@@ -0,0 +1,189 @@
+    // A valid_to in the past is chosen, such that the refunder can refund it immediately
+    let valid_to = chrono::offset::Utc::now().timestamp() as u32 - 1;
+    let quote = OrderQuoteRequest {
+        from: contracts.ethflow.address(),
+        sell_token: contracts.weth.address(),
+        buy_token,
+        receiver,
+        validity: Validity::For(u32::MAX),
+        signing_scheme: QuoteSigningScheme::Eip1271 {
+            onchain_order: true,
+            verification_gas_limit: 0,
+        },
+        side: OrderQuoteSide::Sell {
+            sell_amount: model::quote::SellAmount::AfterFee { value: sell_amount },
+        },
+        ..Default::default()
+    };
+    let quoting = client
+        .post(&format!("{API_HOST}{QUOTING_ENDPOINT}"))
+        .json(&quote)
+        .send()
+        .await
+        .unwrap();
+    assert_eq!(quoting.status(), 200);
+    let quote_response = quoting.json::<OrderQuoteResponse>().await.unwrap();
+    let quote_id: i64 = quote_response.id.unwrap();
+    let fee_amount = quote_response.quote.fee_amount;
+
+    // Creating ethflow order
+    let ethflow_order: EncodedEthflowOrder = (
+        buy_token,
+        receiver.unwrap(),
+        sell_amount,
+        buy_amount,
+        Bytes([0u8; 32]),
+        fee_amount,
+        valid_to,
+        false,
+        quote_id,
+    );
+
+    // Each ethflow user order has an order that is representing
+    // it as EIP1271 order with a different owner and valid_to
+    let technical_order = OrderBuilder::default()
+        .with_kind(OrderKind::Sell)
+        .with_sell_token(contracts.weth.address())
+        .with_sell_amount(sell_amount)
+        .with_fee_amount(fee_amount)
+        .with_receiver(receiver)
+        .with_buy_token(token.address())
+        .with_buy_amount(buy_amount)
+        .with_valid_to(u32::MAX)
+        .with_eip1271(
+            contracts.ethflow.address(),
+            contracts.ethflow.address().0.to_vec(),
+        )
+        .build();
+    let domain_separator = DomainSeparator(
+        contracts
+            .gp_settlement
+            .domain_separator()
+            .call()
+            .await
+            .expect("Couldn't query domain separator")
+            .0,
+    );
+    let order_hash = H256(hashed_eip712_message(
+        &domain_separator,
+        &technical_order.data.hash_struct(),
+    ));
+
+    // Mine ethflow order
+    tx_value!(
+        user,
+        sell_amount + fee_amount,
+        contracts.ethflow.create_order(ethflow_order)
+    );
+
+    // Run autopilot indexing loop
+    services.maintenance.run_maintenance().await.unwrap();
+
+    // Create the refund service and execute the refund tx
+    let pg_pool = PgPool::connect_lazy("postgresql://").expect("failed to create database");
+    let mut refunder = RefundService::new(
+        pg_pool,
+        web3,
+        contracts.ethflow.clone(),
+        -3i64, // Needs to be negative, as valid to was chosen to be in the past
```

Comment:
Just to see if I understand: this needs to be done because the time of the refunder cannot be set arbitrarily. right? Otherwise the test would actually need to wait the time until it expires.